frontend/src/components/object/ObjectFilters.vue

```diff
@@ -0,0 +1,138 @@
+const tagStore = useTagStore();
+const { getMetadataSearchResults } = storeToRefs(useMetadataStore());
+const { getTagSearchResults } = storeToRefs(useTagStore());
+
+// State
+const searching = ref(false);
+const selectedMetadata = ref([]);
+const selectedTags = ref([]);
+
+// Computed
+const metadataValues = computed(() => {
+  // Take the metadata for all objects, distinct, and flatten them into a single array
+  // Add a display property to each tag to be used by the multiselect
+  const distinctVals = [
+    ...new Set(getMetadataSearchResults.value.flatMap((obj) => obj.metadata)),
+  ];
+  return distinctVals
+    .map((val) => ({ ...val, display: `${val.key}:${val.value}` }))
+    .filter(
+      (val, index, self) =>
+        index === self.findIndex((t) => t.display === val.display)
+    )
+    .sort((a, b) => a.display.localeCompare(b.display));
+});
+
+const tagsetValues = computed(() => {
+  // Take the tags for all objects, distinct, and flatten them into a single array
+  // Add a display property to each tag to be used by the multiselect
+  // coms-id not allowed as a tag to filter on by COMS
+  const distinctVals = [
+    ...new Set(getTagSearchResults.value.flatMap((obj) => obj.tagset)),
+  ];
+  return distinctVals
+    .map((val) => ({ ...val, display: `${val.key}=${val.value}` }))
+    .filter((val) => val.key !== 'coms-id')
+    .filter(
+      (val, index, self) =>
+        index === self.findIndex((t) => t.display === val.display)
+    )
+    .sort((a, b) => a.display.localeCompare(b.display));
+});
+
+// Actions
+const selectedFilterValuesChanged = () => {
+  // Get the 'display' property out from selected tag and metadata
+  const metaToSearch: Array<MetadataPair> = selectedMetadata.value.map(
+    ({ ...meta }: any) => meta
+  );
+  const tagSetToSearch: Array<Tag> = selectedTags.value.map(
+    ({ ...tag }: any) => tag
+  );
+
+  // Search the object store with the tagset as a param and metadata as headers
+  objectStore.fetchObjects(
+    {
+      bucketId: props.bucketId,
+    },
+    tagSetToSearch,
+    metaToSearch
+  );
+};
+
+const searchMetadata = async () => {
+  searching.value = true;
+  //await 4 seconds
+  await metadataStore.searchMetadata();
+  searching.value = false;
+};
+
+const searchTagging = async () => {
+  searching.value = true;
+  //await 4 seconds
+  await tagStore.searchTagging();
+  searching.value = false;
+};
+
+</script>
+
+<template>
+  <MultiSelect
+    v-model="selectedMetadata"
+    :options="metadataValues"
+    :loading="searching"
+    class="filterDropdown mr-4"
+    display="chip"
+    option-label="display"
+    placeholder="Metadata"
+    filter
+    filter-placeholder="Search metadata"
+    :show-toggle-all="false"
+    @show="searchMetadata"
+    @update:model-value="selectedFilterValuesChanged"
+  />
+
+  
+  <MultiSelect
+    v-model="selectedTags"
+    :options="tagsetValues"
+    :loading="searching"
+    class="filterDropdown"
+    display="chip"
```

Comment:
![Screenshot 2023-07-13 at 13-26-49 BCBox - My Objects](https://github.com/bcgov/bcbox/assets/1546326/da5b9ce3-d530-4efb-b5b3-39fa7c5f758d)
The chip is a bit large vertically - if there's a compact option available that would be ideal so that it doesn't touch/spill over the bottom field line.

Reply:
Good catch, had to make the dropdown thinner to line up with the search box. Fine tuned the specific sizing for this specific set of multiselect chips